### PR TITLE
Prepare Data Importer for GA [FE-1936]

### PR DESCRIPTION
**Pull request by cleve-fauna**
[FE-1936](https://faunadb.atlassian.net/browse/FE-1936)

### Notes

The Data Importer has been showing some quality issues when tested under load.

This PR does a few things:

- It fixes the correctness bugs
- adds unit tests to the components used
- adds a rate limiter we can sub in with better rate limiting as a 2nd iteration (separate PR)

### Testing

- Have load tested the refactored data importer on 100 files of all types `[csv, scsv, json, jsonl]` on files of 1000-10000 rows and got expected behavior. You can find test utilities here: https://github.com/fauna/testtools/tree/main/DataImportTestTool
- Also ran the load test each type `[csv, scsv, json, jsonl]` on files of 100,000 size and got expected behavior.
- ran the full suite of `scrappy tests` you see in the shell script
- `npm run test` to run the tests defined in our package

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,6 @@ experiments
 .env
 .idea
 /.vscode/
+*~
+*#
+.log
```
Comment (faunaee) [inline .gitignore:14]: Ideally, there would be a newline at the end of this line.

**Review by faunaee — COMMENTED** [review]
A single nit. I haven't tested the code yet, but will try later this afternoon.

**Review by henryfauna — DISMISSED** [review]

File: import_test_data/run_imports.sh [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+# Comment out test batches as required.
+# run_type_tests
+# short_row_tests
+# header_name_tests
+# json_tests
+# invalid_file_type_tests
+directory_tests
+# append_tests
+# directory_specify_collection_tests
+# specify_db_tests
```
Comment (macnealefauna) [inline import_test_data/run_imports.sh:359]: Let's leave these uncommented to start. People should run all the tests by default.

File: import_test_data/run_imports.sh [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+    echo "\033[31m TEST FAILED: $MESSAGE \033[0m" >&2
+    exit 1
+}
+
+run_type_tests () {
+  cleanup_collection "bool_type"
+  $FAUNA_CMD import --endpoint data-import-test --type=favorite::bool --path=type_tests/bool_type.csv
+  if [ $? != 0 ];then
+    fail_test "bool_type.csv didn't import with success"
+  fi
```
Comment (macnealefauna) [inline import_test_data/run_imports.sh:88]: I thought I heard there was some effort to read out data from these tests. I don't see that in this change. Did I miss it?

**Review by macnealefauna — CHANGES_REQUESTED** [review]

File: import_test_data/run_imports.sh (hunk 2 above)
Comment (cleve-fauna) [reply import_test_data/run_imports.sh:359]: oops!

File: import_test_data/run_imports.sh (hunk 3 above)
Comment (cleve-fauna) [reply import_test_data/run_imports.sh:88]: i added a bunch of tests to in the node package itself.

those tests read data and confirm equality

Comment (cleve-fauna) [reply import_test_data/run_imports.sh:88]: i left these as truly scrappy.

File: import_test_data/run_imports.sh (hunk 2 above)
Comment (cleve-fauna) [inline import_test_data/run_imports.sh:359]:
```suggestion
# Comment out test batches as required.
run_type_tests
short_row_tests
header_name_tests
json_tests
invalid_file_type_tests
directory_tests
append_tests
directory_specify_collection_tests
specify_db_tests
```

Comment (cleve-fauna) [reply import_test_data/run_imports.sh]: fixed.

**Review by henryfauna — APPROVED** [review]

**Pull request merged**